src/node_file.cc

```diff
@@ -479,9 +479,9 @@ void FillStatsArray(double* fields, const uv_stat_t* s) {
   fields[9] = -1;
 #endif
   // Dates.
-#define X(idx, name)                          \
-  fields[idx] = (s->st_##name.tv_sec * 1e3) + \
-                (s->st_##name.tv_nsec / 1e6); \
+#define X(idx, name)                                                      \
+  fields[idx] = (static_cast<unsigned long>(s->st_##name.tv_sec) * 1e3) + \
+                (static_cast<unsigned long>(s->st_##name.tv_nsec) / 1e6); \
```

Comment:
I think it would be a good idea if you could leave a comment here, explaining why this is needed.

And just so I get it right; this is needed because `tv_sec` overflows? I think the real problem in that case is that we don’t check for overflows in libuv during `fs__utime_handle`, which we probably should? Am I getting it totally wrong?